dotnet/src/Connectors/Connectors.Redis.UnitTests/RedisHashSetGenericDataModelMapperTests.cs

```diff
@@ -0,0 +1,206 @@
+
+        // Assert.
+        Assert.Equal("key", dataModel.Key);
+        Assert.Equal("data 1", dataModel.Data["StringData"]);
+        Assert.Equal(1, dataModel.Data["IntData"]);
+        Assert.Equal(2u, dataModel.Data["UIntData"]);
+        Assert.Equal(3L, dataModel.Data["LongData"]);
+        Assert.Equal(4ul, dataModel.Data["ULongData"]);
+        Assert.Equal(5.5d, dataModel.Data["DoubleData"]);
+        Assert.Equal(6.6f, dataModel.Data["FloatData"]);
+        Assert.True((bool)dataModel.Data["BoolData"]!);
+        Assert.Equal(7, dataModel.Data["NullableIntData"]);
+        Assert.Equal(8u, dataModel.Data["NullableUIntData"]);
+        Assert.Equal(9L, dataModel.Data["NullableLongData"]);
+        Assert.Equal(10ul, dataModel.Data["NullableULongData"]);
+        Assert.Equal(11.1d, dataModel.Data["NullableDoubleData"]);
+        Assert.Equal(12.2f, dataModel.Data["NullableFloatData"]);
+        Assert.False((bool)dataModel.Data["NullableBoolData"]!);
+        Assert.Equal(new float[] { 1, 2, 3, 4 }, ((ReadOnlyMemory<float>)dataModel.Vectors["FloatVector"]!).ToArray());
+        Assert.Equal(new double[] { 5, 6, 7, 8 }, ((ReadOnlyMemory<double>)dataModel.Vectors["DoubleVector"]!).ToArray());
+    }
+
+    [Fact]
+    public void MapFromStorageToDataModelMapsNullValues()
+    {
+        // Arrange
+        VectorStoreRecordDefinition vectorStoreRecordDefinition = new()
+        {
+            Properties = new List<VectorStoreRecordProperty>
+            {
+                new VectorStoreRecordKeyProperty("Key", typeof(string)),
+                new VectorStoreRecordDataProperty("StringData", typeof(string)),
+                new VectorStoreRecordDataProperty("NullableIntData", typeof(int?)),
+                new VectorStoreRecordVectorProperty("FloatVector", typeof(ReadOnlyMemory<float>?)),
+            },
+        };
+
+        var hashSet = new HashEntry[]
+        {
+            new("storage_string_data", RedisValue.Null),
+            new("NullableIntData", RedisValue.Null),
+            new("FloatVector", RedisValue.Null),
+        };
+
+        var sut = new RedisHashSetGenericDataModelMapper(RedisHashSetVectorStoreMappingTestHelpers.s_vectorStoreRecordDefinition);
+
+        // Act
+        var dataModel = sut.MapFromStorageToDataModel(("key", hashSet), new() { IncludeVectors = true });
+
+        // Assert
+        Assert.Equal("key", dataModel.Key);
+        Assert.Null(dataModel.Data["StringData"]);
+        Assert.Null(dataModel.Data["NullableIntData"]);
+        Assert.Null(dataModel.Vectors["FloatVector"]);
+    }
+
+    [Fact]
+    public void MapFromDataToStorageModelSkipsMissingProperties()
+    {
+        // Arrange.
+        var sut = new RedisHashSetGenericDataModelMapper(RedisHashSetVectorStoreMappingTestHelpers.s_vectorStoreRecordDefinition);
+        var dataModel = new VectorStoreGenericDataModel<string>("key")
+        {
+            Data = { },
```

Comment:
nit?

```suggestion
            Data = [],
```

Reply:
These are actually dictionaries, not arrays, so the {} seems more appropriate I think.